client/interledger/js/components/asset_row.js

```diff
@@ -9,87 +12,177 @@ import AssetDetail from '../../../lib/js/react/components/asset_detail';
 
 const AssetRow = React.createClass({
     propTypes: {
+        account: React.PropTypes.object.isRequired,
+        asset: React.PropTypes.object.isRequired,
         accountList: React.PropTypes.array,
-        activeAccount: React.PropTypes.object,
-        asset: React.PropTypes.object,
+        actionMap: React.PropTypes.object,
         handleAccountClick: React.PropTypes.func,
         handleAssetClick: React.PropTypes.func,
         isActive: React.PropTypes.bool
     },
 
+    getDefaultProps() {
+        return {
+            actionMap: {
+                'single-owner-transfer': {
+                    actionName: 'ESCROW',
+                    actionMessage: 'Escrow asset with:',
+                    selectAccounts: true
+                },
+                'multi-owner-execute': {
+                    actionName: 'EXECUTE',
+                    actionMessage: 'Execute escrow of asset:',
+                    selectAccounts: false
+                },
+                'multi-owner-abort': {
+                    actionName: 'ABORT',
+                    actionMessage: 'Abort escrow of asset:',
+                    selectAccounts: false
+                }
+            }
+        };
+    },
+
     getInitialState() {
         return {
-            inEscrow: false
+            expiresIn: null
         };
     },
 
+    componentDidMount() {
+        if (this.getOperation() !== 'transfer') {
+            this.intervalId = window.setInterval(() => {
+                this.setExpiryTime();
+            }, 1000);
+        }
+    },
+
+    componentWillUnmount() {
+        window.clearInterval(this.intervalId);
+    },
+
+    setExpiryTime() {
+        const {
+            asset
+        } = this.props;
+        const expires = moment(parseInt(parseFloat(asset.expiryTime) * 1000, 10));
```

Comment:
Not sure if you need both the `parseFloat` and `parseInt`; in JS all numbers are doubles. Seeing [moment's unix parsing](http://momentjs.com/docs/#/parsing/unix-timestamp/), it's probably fine just to use `moment.unix(parseFloat(asset.expiryTime))`, assuming `asset.expiryTime` is not just an integer.